editor/block-mover/generate-order-title.js

```diff
@@ -0,0 +1,21 @@
+export default function( { typeTitle, position, isFirst, isLast, dir } ) {
+	const prefix = 'Move "' + typeTitle + '" block from position ' + position;
+	const errorPrefix = 'Block "' + typeTitle + '"';
+	let title = '';
+
+	if ( isFirst && isLast ) {
+		title = errorPrefix + ' is the only block, and cannot be moved';
+	} else if ( dir > 0 ) {
+		// moving down
+		title = ( ! isLast ) ?
+			prefix + ' down to position ' + ( position + 1 ) :
+			errorPrefix + ' is at the end of the content and can’t be moved down';
+	} else {
+		// moving up
+		title = ( ! isFirst ) ?
+			prefix + ' up to position ' + ( position - 1 ) :
+			errorPrefix + ' is at the beginning of the content and can’t be moved up';
+	}
+
+	return title;
```

Comment:
Have you considered using an _early-abort_ style instead of doing more nesting and mutation of `title`?

```js
if ( isFirst && isLast ) {
	return `${ errorPrefix } is the only block and cannot be moved';
}

// moving down and last block
if ( dir > 0  && isLast ) {
	return `${ errorPrefix } is at the beginning of the content and can't be moved up`;
}

// moving down and not first block
if ( dir > 0 && ! isLast ) {
	return `${ prefix } down to position ${ position + 1 }`;
}

// moving up and first block
if ( dir < 0 && isFirst ) {
	return `${ errorPrefix } … `
}

// moving up and not first block
if ( dir < 0 && ! isFirst ) {
	return `${ prefix }…`;
}

// no other valid options, return empty
return '';
```